node/src/main.rs

```diff
@@ -373,13 +374,37 @@ async fn main() {
         if !opt.disable_block_ingestor {
             if ethereum_chains.len() > 0 {
                 let block_polling_interval = Duration::from_millis(opt.ethereum_polling_interval);
+                let (firehose, polling): (HashMap<_, _>, HashMap<_, _>) = ethereum_chains
+                    .into_iter()
+                    .partition(|(_, chain)| chain.is_firehose_supported());
 
-                start_block_ingestor(
-                    &logger,
-                    &logger_factory,
-                    block_polling_interval,
-                    ethereum_chains,
-                );
+                start_block_ingestor(&logger, &logger_factory, block_polling_interval, polling);
```

Comment:
Should we avoid completely the RPC block ingestor if Firehose is enabled? The Firehose one will have better latency for new blocks in general, so seems a fair assumption to use it exclusively if an endpoint for it exist.

Reply:
This behaviour is implicit because the IS_FIREHOSE_PREFERRED env var is not set per chain. This means that if it's set and there are firehose endpoints, then firehose is used for ingestor. On the other hand if a specific chain does not have firehose endpoints it needs to run RPC

Reply:
We also can't remove this completely because some ethereum compatible chains may exist and use the RPC ingestor

Reply:
The way I'm reading the current code for when both an RPC and a Firehose provider are set is the following (please correct any wrong assumptions about the behavior of your PR):

  ```
    ...
    provider = [
      { label = "firehose", details = { type = "firehose", url = "<url>", token = "<auth_token>", features = [ "filters" ] }},
      { label = "peering", url = "<ethereum_full_node_rpc>", features = [ "archive" ] },
    ]
  ```
  
In this case, `start_block_ingestor` is called which will launch the current `EthereumBlockIngestor`. And since there is a Firehose provider configured for Ethereum, `start_firehose_block_ingestor` will also be called.

I'm commenting on this specific thing where only one of EthereumBlockIngestor or `FirehoseBlockIngestor` must started for a given chain.